src/main/java/ecommerce/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,44 @@
+package ecommerce.service.impl;
+
+import ecommerce.dao.OrderDao;
+import ecommerce.lib.Inject;
+import ecommerce.lib.Service;
+import ecommerce.model.Order;
+import ecommerce.model.ShoppingCart;
+import ecommerce.service.OrderService;
+import java.util.List;
+
+@Service
+public class OrderServiceImpl implements OrderService {
+
+    @Inject
+    OrderDao orderDao;
+
+    @Override
+    public Order completeOrder(ShoppingCart shoppingCart) {
+        Order order = orderDao.completeOrder(shoppingCart);
+        order.setProducts(shoppingCart.getProducts());
+        shoppingCart.getProducts().removeAll(shoppingCart.getProducts());
```

Comment:
We can use clear() here.
Also after changing any object (shoppingCart in this case) we need to invoke update() method to save data to DB to my mind.

Reply:
done